docs/client-guidelines.md

```diff
@@ -1,11 +1,89 @@
 # Client writing guidelines
 
-Unsure what else should go here, please add thoughts if you end up writing a client.
+The clients should default to using `TRAFFICLIGHT_URL` to represent the server the trafficlight client should connect to for information. This should contain the base URL of the trafficlight server (without `/client`).
 
-The clients should default to using `TRAFFICLIGHT_URL` to represent the server the trafficlight client
-should connect to for information. This should contain the base URL of the trafficlight server (without `/client`).
+## Client behaviour
 
-Clients should aim to poll at least every 60s and should ideally respect the delay parameter in the `idle` respnose.
-This might become more handwavey if the client is performing a blocking action where it's unable to poll while working.
+Clients should register then immediately start polling for actions to perform.
 
+Clients that have not been allocated a test case will be dropped if they do not poll at least every 60s. 
 
+Clients that have been allocated a test case will cause a timeout and fail the test, if they do not poll or respond at least every 180s.
+
+If your action is long-running, ideally split it into smaller components internally and trigger extra polls of the server between them. You should take appropriate steps if the action changes (eg an action of "exit" should stop the test).
+
+Do not have a background task that polls the server for an action reguardless of progress - we want the timeout to occur if no progress occurs in a fixed period of time.
+
+## Action naming
+
+Action should be named in `snake_case` - no capitals, spaces replaced by underscores.
+
+We chose this because the tests are written in python and that's the native practice for naming methods. This means the actions in the test case code will be named the same as the actions on the wire.
+
+We Should name actions with a verb at the start and what is being acted on later. The test should aim to not require comments to understand what is going on at each step.
+ * `create_new_room("test_room_name")`
+ * `invite_user("bob:server2")`
+
+We should reserve `get` for passive actions that obtain information about what the client sees. 
+ * `get_timeline`
+ * `get_all_devices`
+
+We Should Not name actions passively, or indirectly
+ * `invite_user` instead of `do_the_user_invite`
+ * `crosssign_user` instead of `invoke_crosssign`
+
+We Should name actions after what they do, not what test they're related to
+ * `create_poll` not `pr_52_action` or `msc_1235_action`
+
+We Should avoid naming actions in a client specific way
+ * `clear_idb_cache` is not great because it's only available in element web
```

Comment:
This isn't client specific but rather platform specific. I imagine similar actions would be needed for other platforms as well.

Reply:
Have added some possibly clearer negative examples around mouse hovering etc.